### PR TITLE
GEODE-733: Cluster Config service can throw a spurious exception

### PR DESCRIPTION
**Pull request by alb3rtobr**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [n/a] Have you written or updated unit tests to verify your changes?

- [n/a] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/test/java/org/apache/geode/distributed/internal/InternalConfigurationPersistenceServiceTest.java [hunk 1]
```diff
@@ -265,6 +270,26 @@ public void removeInvalidXmlConfiguration(String xml, int expectedInitialElement
...
+    InternalDistributedSystem.Builder b =
+        new InternalDistributedSystem.Builder(new Properties(), metricsSessionBuilder);
+    InternalDistributedSystem ds = b.build();
+
+    service3 = spy(new InternalConfigurationPersistenceService(
+        DistributedLockService.create("test", ds),
+        JAXBService.createWithValidation(CacheConfig.class)));
+    doReturn(false).when(service3).lockSharedConfiguration();
+
+    service3.createConfigurationResponse(null);
```
Comment (alb3rtobr) [inline geode-core/src/test/java/org/apache/geode/distributed/internal/InternalConfigurationPersistenceServiceTest.java]: Without the code changes, this test fails in line 289. As Im mocking the call to `lockSharedConfiguration()` to return `false`, an exception is thrown when trying to unlock shared config inside `createConfigurationResponse`

File: geode-core/src/test/java/org/apache/geode/distributed/internal/InternalConfigurationPersistenceServiceTest.java [hunk 2]
```diff
@@ -265,6 +270,26 @@ public void removeInvalidXmlConfiguration(String xml, int expectedInitialElement
         .isEqualTo(expectFinalElements);
   }
 
+  @Test
+  public void dontUnlockSharedConfigurationIfNoLockedPreviously() {
+    InternalConfigurationPersistenceService service3;
+
+    MetricsService.Builder metricsSessionBuilder = mock(MetricsService.Builder.class);
```
Comment (jinmeiliao) [inline geode-core/src/test/java/org/apache/geode/distributed/internal/InternalConfigurationPersistenceServiceTest.java]: looks like you don't need all these code to build ds, and you also don't need that extra "visisble for testing" constructor you added. You can just do: 


    JAXBService jaxbService = mock(JAXBService.class);
    InternalConfigurationPersistenceService cps = spy(new InternalConfigurationPersistenceService(jaxbService));
    doReturn(false).when(cps).lockSharedConfiguration();
    cps.createConfigurationResponse(null);
......

Comment (alb3rtobr) [reply geode-core/src/test/java/org/apache/geode/distributed/internal/InternalConfigurationPersistenceServiceTest.java]: you are right, that simplifies a lot the test case. Thanks!

**Comment by jinmeiliao** [comment]
@alb3rtobr please resolve the conflict before I can merge this changeset. Thanks!

**Comment by alb3rtobr** [comment]
@jinmeiliao Done!

**Pull request merged**